p/api/greader.php

```diff
@@ -917,6 +921,12 @@ function markAllAsRead($streamId, $olderThanId) {
 if ($user !== '') {
 	FreshRSS_Context::$user_conf = get_user_configuration($user);
 	Minz_Translate::init(FreshRSS_Context::$user_conf->language);
+
+	Minz_ExtensionManager::init();
+
+	if (FreshRSS_Context::$user_conf != null) {
+		Minz_ExtensionManager::enableByList(FreshRSS_Context::$user_conf->extensions_enabled);
+	}
 } else {
```

Comment:
Would `$system_conf->allow_anonymous)` be relevant to the GReader API?

Reply:
It seems relevant, I guess (even if I wouldn't have the usage, personally), but it's a bit out of context of this PR, no ? (or I didn't understand the point…).

Reply:
If it's relevant then the extension hook is relevant, along these lines:

https://github.com/FreshRSS/FreshRSS/blob/7802fd33a627dc7c582df871cfa613d9be8f8788/app/FreshRSS.php#L51-L55

Reply:
Basically I mean that this else path seems to be about users who aren't logged in. I haven't looked into this, mind you, I'm just looking at the diff here.

Reply:
Okay, I see. I think it’s a bit out of my understanding of the sources :s

How `FreshRSS_Context::$user_conf` can't be null if there is no `$user` ? We have to find the default user, in this case ? Or `$user_conf` already point to the default user before we set it to NULL (by security, perhaps ?) ?

I think I don't know the code enough (and the consequences) to play around that part…

That being said, apparently the `allow_anonymous` is not handled at all by greader, it just fails if a user is not set & valid, so the question of extension loading in this condition will have to be handled when `allow_anonymous` will be handled in a whole for greader, no ?

```
if ($user == '') {
		unauthorized();
	}
```

Reply:
That means the answer to my question would be that it's *not* in fact relevant. ;-)

Reply:
> Basically I mean that this else path seems to be about users who aren't logged in. I haven't looked into this, mind you, I'm just looking at the diff here.

You mean
```
} else {
	Minz_Translate::init();
}
```
?

Yes, I'm not sure why it was done, greader doesn't seems to allow anything, currently, if there is no authenticated user.

Reply:
Yes, that's why I commented on the else line. That wasn't just a random comment, or I wouldn't have attached it to a line. ^_^

Reply:
> That would mean the answer to my question is that it's _not_ relevant. ;-)

Ok, yes. I better understand your point now.

It **would** be relevant to load default user extensions if `allow_anonymous` was handled in a whole by this API, but as it doesn't seems to be the case, then I guess it's not relevant, at least for now.